packages/inferno-generator/src/expressions/inferno-component.ts

```diff
@@ -193,20 +193,22 @@ export class InfernoComponent extends PreactComponent {
     const createEffectsStatements: string[] = [];
     const updateEffectsStatements: string[] = [];
     if (this.effects.length || this.jQueryRegistered) {
-      const dependencies = this.effects.map(
+      const dependenciesString = this.effects.map(
         (e) => e.getDependency(this.getToStringOptions())
-          .filter((dep) => {
-            const depMember = this.getToStringOptions().members.find(
-              (member) => member.name === dep,
-            );
-            return !depMember?.isMutable;
+          .filter((dep) => (dep instanceof BaseClassMember ? !dep.isMutable : true))
```

Comment:
I suggest to remove the filter and move !dep.isMutable in to the map if expression